Examples/Nodify.Playground/BaseSettingViewModel.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace Nodify.Playground
+{
+    public class BaseSettingViewModel<T> : ObservableObject, ISettingViewModel
+    {
+        public string Name { get; }
+        public string? Description { get; } // This is the tooltip
+
+        private object? _value;
+
+        object? ISettingViewModel.Value
+        {
+            get => _value;
+            set => SetProperty(ref _value, value);
+        }
+
+        public SettingsType Type { get; set; }
+
+        public T Value
+        {
+            get => ((ISettingViewModel)this).Value is T val ? val : default!;
```

Comment:
Let's assume that the Value is always of type T, and if that's not true, we want to fail fast to avoid potential bugs. (cast to T directly)